### DISCUSSION TITLE
Swift Compiler Error (Xcode): Compiling for iOS 12.0, but module 'FirebaseAuth' has a minimum deployment target of iOS 13.0

### DISCUSSION THREAD

**Discussion by menaghaly** [codemagic-ci-cd/codemagic-docs#2727] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. added in podfile platform :ios, '14.0'
2. it was building fine but when added in Appdelegate.swift Firebase it stopped building

### Expected results

should build correctly 

### Actual results

gave me this error 
Failed to build iOS app
Swift Compiler Error (Xcode): Compiling for iOS 12.0, but module 'FirebaseAuth' has a minimum deployment target of iOS 13.0: /Users/builder/Library/Developer/Xcode/DerivedData/Runner-edaimyiflreloheqntgnhkmwcclv/Build/Intermediates.noindex/ArchiveIntermediates/Runner/BuildProductsPath/Release-iphoneos/FirebaseAuth/FirebaseAuth.framework/Modules/FirebaseAuth.swiftmodule/arm64-apple-ios.swiftmodule

Encountered error while archiving for device.

Build failed :|
Failed to build for iOS

### Build id (optional)

66dfec23d262df12062594cf

**Comment by dtrdic** [comment]
hey @menaghaly you can modify your Podfile to target iOS 13 by adding  `config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '13.0'`
or you can target iOS 13 for specific pods like for FirebaseAuth, while keeping the rest of your project compatible with iOS 12 (or whatever your preferred target is).
You can do this by adding logic inside the post_install block.
```
post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)

    # Specify iOS 13.0 as the minimum deployment target for FirebaseAuth
    if ['FirebaseAuth'].include?(target.name)
      target.build_configurations.each do |config|
        config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '13.0'
      end
    end
  end
```

**Reply by menaghaly** [reply]
still gives me the same error Failed to build iOS app
Swift Compiler Error (Xcode): Compiling for iOS 12.0, but module 'FirebaseAuth' has a minimum deployment target of iOS 13.0: /Users/builder/Library/Developer/Xcode/DerivedData/Runner-edaimyiflreloheqntgnhkmwcclv/Build/Intermediates.noindex/ArchiveIntermediates/Runner/BuildProductsPath/Release-iphoneos/FirebaseAuth/FirebaseAuth.framework/Modules/FirebaseAuth.swiftmodule/arm64-apple-ios.swiftmodule

Encountered error while archiving for device.


Build failed :|
Failed to build for iOS
 and this is my podfile 
# Uncomment this line to define a global platform for your project
platform :ios, '14.0'

# CocoaPods analytics sends network stats synchronously affecting flutter build latency.
ENV['COCOAPODS_DISABLE_STATS'] = 'true'

project 'Runner', {
  'Debug' => :debug,
  'Profile' => :release,
  'Release' => :release,
}

def flutter_root
  generated_xcode_build_settings_path = File.expand_path(File.join('..', 'Flutter', 'Generated.xcconfig'), __FILE__)
  unless File.exist?(generated_xcode_build_settings_path)
    raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
  end

  File.foreach(generated_xcode_build_settings_path) do |line|
    matches = line.match(/FLUTTER_ROOT\=(.*)/)
    return matches[1].strip if matches
  end
  raise "FLUTTER_ROOT not found in #{generated_xcode_build_settings_path}. Try deleting Generated.xcconfig, then run flutter pub get"
end

require File.expand_path(File.join('packages', 'flutter_tools', 'bin', 'podhelper'), flutter_root)

flutter_ios_podfile_setup

target 'Runner' do
  use_frameworks!
  use_modular_headers!
  # Add Google Maps SDK
  pod 'GoogleMaps'
  pod 'Firebase/Core'
  pod 'Firebase/Messaging'

  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))

  target 'RunnerTests' do
    inherit! :search_paths
  end
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)

    if ['FirebaseAuth'].include?(target.name)
      target.build_configurations.each do |config|
        config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '13.0'

        # Ensure Swift version is set for compatibility with Apple Sign-In
        config.build_settings['SWIFT_VERSION'] = '5.10'
      end
    end
  end
end

**Reply by dtrdic** [reply]
Please try removing the Pods directory and the Podfile.lock file, as Podfile.lock keeps track of the pod versions installed during the last pod install. If there are changes in your Podfile (such as updating the deployment target or dependencies), they may not be fully reflected until the Podfile.lock is regenerated

```
rm Podfile.lock
rm -rf Pods
pod install
```

**Reply by menaghaly** [reply]
I'm working on a windows not a mac using visual studio code for editing, so how can i do that ?

**Reply by dtrdic** [reply]
you don't have to run `pod instal`l then, removing Podfile.lock and Pods folder should be enough. Pod install will run on the build machine anyway

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
